tests/virt_autotest/virt_utils.pm

```diff
@@ -81,20 +81,39 @@ sub locate_sourcefile {
 }
 
 sub get_repo_0_prefix {
-    # Get customized repo location from REPO_0_PREFIX and append missing forward slash to the end
+    my ($default_repo_urls_ref) = @_;
+    my @default_repo_urls = @{$default_repo_urls_ref};
     my $repo_0_prefix = get_var("REPO_0_PREFIX", "");
     $repo_0_prefix .= '/' unless $repo_0_prefix =~ /\/$/;
-    $repo_0_prefix = ($repo_0_prefix ne "/" ? $repo_0_prefix : "http://openqa.suse.de/assets/repo/");
-    return $repo_0_prefix;
+    foreach my $url (@default_repo_urls) {
+        my $repo_0 = get_var("REPO_0");
+        record_info("REPO_0 Original", "Original value of REPO_0: $repo_0");
+        my $version = get_version_for_daily_build_guest;
+        record_info("SUSE Version Check", "Checking if SUSE version ($version) requires modifying REPO_0.");
+        $repo_0 =~ s/-Online-/-Full-/ if ($version =~ /15-sp[2-9]/i);
+        record_info("REPO_0 Modified", "Modified value of REPO_0: $repo_0");
+        my $repo_0_prefix = ($repo_0_prefix ne "/" ? $repo_0_prefix : "http://$url/assets/repo/");
+        record_info("Final REPO_0_PREFIX", "Final constructed REPO_0_PREFIX URL: $repo_0_prefix");
+        # Check the HTTP response for this URL
+        my $check_url = "$repo_0_prefix$repo_0/CHECKSUMS";
+        record_info("URL Check", "Checking accessibility of URL: $check_url");
+        my $result = script_run("curl -sSf '$check_url' > /dev/null");
+        if ($result == 0) {
+            record_info("URL Accessible", "Successfully accessed URL: $check_url");
+            return $repo_0_prefix;    # Return the first accessible URL
+        }
+    }
+    die "All provided repo URLs are inaccessible.";
 }
 
 sub repl_repo_in_sourcefile {
     # Replace the daily build repo as guest installation resource in source file (like source.cn; source.de ..)
     my $verorig = "source.http.sles-" . get_version_for_daily_build_guest . "-64";
     my $veritem = is_x86_64 ? $verorig : get_required_var('ARCH') . ".$verorig";
     if (get_var("REPO_0")) {
+        my @repo_urls = ('openqa.suse.de', 'openqa.qa2.suse.asia');
```

Comment:
@RoyCai7  If I understand correctly, you'd like to solve the problem of tested repository being used in BJ openqa  deleted on OSD? If yes, I think it is enough to replace this line with `my @repo_urls = (get_required_var('OPENQA_URL'));`. 

Reasons:
- when our arm tests run on BJ openqa, the repository won't be deleted since there is space clearning tool applied here, and it will be faster to use local repo
- when the arm tests run on OSD, the space cleaning tool has  a mechanism to NOT delete repo being used by running tests, so you won't loose the needed repo , what's more there won't be arm repo if VT tests are not in BJ openqa

HDYT?

Reply:
I think it is enough to use local repo directly.
@waynechen55 what do you think of it? a little different from your opinion.
Don't use openqa.suse.de, only use openqa.qa2.suse.asia. because local repo is much faster and keep it always there.

Reply:
- Basically you can decide which repo to use based on your observation. But switching to use full media on BJ openQA will also incur downloading the full media every time and mounted it locally manually, because it is full media we also need to maintain it to make sure space on BJ openQA used efficiently.  Advantage of using local media might be speed. I am not sure how fast it could be.
- Advantage of still using full media on OSD is the media just shows up when a build is published and no more extra effort on our side. And we also switch to use local one if something unexpected happened. Flexible.
- Either case, I think we should keep the detect logci, let test run fail immediately if media is not accessible.

Reply:
@RoyCai7 
If we actually do not download arm image each time in BJ, it makes sense to try OSD too.

Our arm tests will be moved back to OSD if we get new arm servers there. Then for that case -- run test on OSD, do we still need to keep BJ image check here? Will you make code ready for both cases -- run test in bj or osd?